src/successModal/index.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { Component } from 'react'
+import { canUseDOM, canUseEventListeners } from 'exenv'
+import { createPortal } from 'react-dom'
+import TransitionGroup from 'react-transition-group/TransitionGroup'
+import CustomTransition, { AnimationType } from 'transitions'
+import { color } from '_utils/branding'
+
+import cc from 'classcat'
+import Button from 'button'
+import style from './style'
+
+const KEYCODES = {
+  ESCAPE: 27,
+}
+
+export interface SuccessModalProps {
+  readonly isOpen?: boolean
+  readonly children?: JSX.Element | JSX.Element[]
+  readonly className?: Classcat.Class
+  readonly closeOnEsc?: boolean
+  readonly onConfirm: () => void
+  readonly confirmLabel?: string
+  readonly illustartionSrc: string
+  readonly illustartionText?: string
+  readonly large?: boolean
+}
+
+class SuccessModal extends Component<SuccessModalProps> {
+  private portalNode: HTMLElement
+
+  static defaultProps: Partial<SuccessModalProps> = {
+    isOpen: false,
+    closeOnEsc: false,
+    large: false,
+    illustartionText: '',
+  }
+
+  constructor(props: SuccessModalProps) {
+    super(props)
+    if (canUseDOM) {
+      this.portalNode = document.createElement('div')
+      document.body.appendChild(this.portalNode)
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.isOpen) {
+      this.addListeners()
+    }
+  }
+
+  componentDidUpdate(prevProps: SuccessModalProps) {
+    if (!prevProps.isOpen && this.props.isOpen) {
+      this.addListeners()
+    }
+
+    if (!this.props.isOpen && prevProps.isOpen) {
+      this.removeListeners()
+    }
+  }
+
+  componentWillUnmount() {
+    this.removeListeners()
+    document.body.removeChild(this.portalNode)
+    this.portalNode = null
+  }
+
+  addListeners() {
+    if (this.props.isOpen && canUseEventListeners) {
+      if (this.props.closeOnEsc) {
+        document.addEventListener('keydown', this.handleKeydown)
+      }
+    }
+  }
+
+  removeListeners() {
+    if (!this.props.isOpen && canUseEventListeners) {
+      document.removeEventListener('keydown', this.handleKeydown)
+    }
+  }
+
+  handleKeydown = (event: KeyboardEvent) => {
+    if (event.keyCode === KEYCODES.ESCAPE) {
+      this.props.onConfirm()
+    }
+  }
+
+  render() {
+    const {
+      isOpen,
+      children,
+      className,
+      large,
+      onConfirm,
+      confirmLabel,
+      illustartionSrc,
+      illustartionText,
+    } = this.props
+
+    const baseClassName = 'kirk-successModal'
+
+    const classNames = cc([
+      baseClassName,
+      {
+        [`${baseClassName}--large`]: large,
+      },
+      className,
+    ])
+
+    const successModalElement = (
+      <div>
+        <TransitionGroup component="div" className="transition-wrapper">
+          {isOpen && (
+            <CustomTransition animationName={AnimationType.FADE}>
+              <div className={classNames}>
+                <div className={`${baseClassName}-dialog`}>
+                  <div className={`${baseClassName}-body`}>
```

Comment:
Try importing the Modal component here instead, and you might be able to remove all the lifecycle methods.

Reply:
Done.